expression/src/lib.rs

```diff
@@ -0,0 +1,51 @@
+// configure default clippy lints
+#![deny(clippy::correctness)]
+#![warn(clippy::complexity, clippy::style, clippy::perf, clippy::pedantic)]
+// disable some pedantic lints
+#![allow(
+    clippy::cast_possible_truncation,
+    clippy::cast_possible_wrap,
+    clippy::cast_precision_loss,
+    clippy::cast_sign_loss,
+    clippy::default_trait_access,
+    clippy::missing_errors_doc,
+    clippy::module_name_repetitions,
+    clippy::must_use_candidate,
+    clippy::non_ascii_literal,
+    clippy::option_if_let_else,
+    clippy::result_large_err, // TODO: investigate this
```

Comment:
whats this?

Reply:
I copied these lints from our other packages, it is identical everywhere.

Result large err says that in Result<A, B>, B shouldn't get that large.
But for us, it does since we store some stack data in the source field.
The workaround would be to Box the error sources, but we have to discuss this sometime :tm:.